zerver/tests/test_openapi.py

```diff
@@ -276,6 +287,68 @@ def check_for_non_existant_openapi_endpoints(self) -> None:
                 msg += "\n + {}".format(undocumented_path)
             raise AssertionError(msg)
 
+    def _highest_priority_type(self, types: List[type]) -> type:
+        priority = {list: 1, str: 2, int: 3, bool: 4}
+        tyiroirp = {1: list, 2: str, 3: int, 4: bool}
+        val = 5
+        for t in types:
+            v = priority.get(t, 5)
+            if v < val:
+                val = v
+        return tyiroirp.get(val, types[0])
+
+    def _get_standardized_argument_type(self, t: Any) -> type:
+        origin = getattr(t, "__origin__", None)
+        if not origin:
+            # Then it's most likely one of the fundamental data types
+            # I.E. Not one of the data types from the "typing" module.
+            return t
+        elif origin == Union:
+            subtypes = []
+            for st in t.__args__:
+                subtypes.append(self._get_standardized_argument_type(st))
+            return self._highest_priority_type(subtypes)
+        elif origin == List:
+            return list
+        return self._get_standardized_argument_type(t.__args__[0])
+
+    def check_argument_types(self, function: Callable[..., HttpResponse],
+                             openapi_parameters: List[Dict[str, Any]]) -> None:
+        op = set([(element["name"], VARMAP[element["schema"]["type"]]) for
+                  element in openapi_parameters])
+
+        fp = set()
+        while getattr(function, "__wrapped__", None):
+            function = function.__wrapped__  # type: ignore # mypy doesn't recognize the condition
+        for vname, defval in inspect.signature(function).parameters.items():
+            defval = defval.default
+            if defval.__class__ == REQ:
+                vtype = self._get_standardized_argument_type(function.__annotations__[vname])
+                vname = defval.post_var_name  # type: ignore # it's a REQ; account for whence
+                fp.add((vname, vtype))
+
+        diff = op - fp
+        if diff:  # nocoverage
+            # Print a *VERY* clear and verbose error message.
+            msg = "There seems to be a mismatch between the types for the request parameters \
+documented in the OpenAPI documentation and the actual request parameters in the function \
+decleration for {}.\n".format(function.__name__)
+            msg += '='*65 + '\n'
+            msg += "{:<10s}{:^30s}{:>10s}\n".format("Parameter", "OpenAPI Type", "Function Decleration Type")
```

Comment:
"Declaration" is mispelled here and in the `msg` definition above.  You may also want to use ` """` heredoc to format the `msg` string better.

Reply:
I fixed the spelling, but I'm not sure about how format the msg string better using multiline strings (heredocs). Are you referring to the new f-strings syntax (I'm guessing not since that's only available on Python 3.6+)?

Reply:
msg = """
There seems to be a mistmatch between 
...
""".format(...)